### PR TITLE
Some functional tests for slappy

### PR DESCRIPTION
**Pull request by pglass**
- This adds two dockerfiles: one with just bind (the "master") and one with bind + slappy (the "slappy"). Each docker container has a corresponding makefile to build, start, stop, and inspect.
- There's a top-level makefile `tests/Makefile` that can manage both docker containers.
- There are test cases for slappy stats and for CUD-ing zones via slappy.
- There are meta tests for utilities to manage zone files on the master.
- Travis is working.
- Tests are run with `tox`.

### CODE REVIEW

**Comment by TimSimmons** [comment]
This is so :fire: I want to pull it down tomorrow and play with it a bit, but it looks good. I can write some more tests to cover a few of the negative cases.

**Comment by ionrock** [comment]
I'm going to :+1: even though I didn't read all the code that carefully. I trust you @pglass.

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+Since these are functional tests, we need actual bind nameservers to test
+against. This is accomplished with two docker containers. One container is
+the master (running just bind), and the other container is the target (running
+both slappy and bind).
+
+You'll need to:
+
+1. Install docker
+2. Install go
```
Comment (TimSimmons) [inline tests/README.md]: For some reason I didn't have `jq` installed, and that's used in `slappy.makefile`.

**Comment by TimSimmons** [comment]
This is great. Merge away :shipit: 

I have a patch or two sitting in tree, and I need to fix #33 soon. When I submit all that, I'll probably add a bit more documentation and maybe a couple more Makefile targets to make it even more magic. I want to write some more tests too. But I really like this strategy. I'm containering out right now.
